### PR TITLE
Reference extension maturity matrix from extensions FAQ

### PR DESCRIPTION
**Pull request by holly-cummins**
Now that the maturity matrix is live, we can cross-reference it from the extensions FAQ.

### CODE REVIEW

File: docs/src/main/asciidoc/extension-faq.adoc [hunk 1]
```diff
@@ -19,8 +19,8 @@ The document header ends at the first blank line. Do not remove the blank line b
 ### Why would I want to write an extension?
 
 See the xref:writing-extensions#extension-philosophy[extension philosophy].
-
-One useful thing extensions can do is bundle other extensions.
+The xref:extension-maturity-matrix[extension maturity matrix] shows the kinds of capabilities extensions can offer.
```
Comment (gsmet) [inline docs/src/main/asciidoc/extension-faq.adoc:22]:
```suggestion
The xref:extension-maturity-matrix.adoc[extension maturity matrix] shows the kinds of capabilities extensions can offer.
```

File: docs/src/main/asciidoc/extension-faq.adoc [hunk 2]
```diff
@@ -30,6 +30,11 @@ Not every problem needs an extension!
  For example, plain libraries can create new configuration elements and register classes with Jandex (this link:https://www.loicmathieu.fr/wordpress/en/informatique/quarkus-tip-comment-ne-pas-creer-une-extension-quarkus/[blog shows how]).
 
 
+## How do I know what kind of capabilities I might want to include in an extension?
+
+Have a look at the xref:extension-maturity-matrix[extension maturity matrix].
```
Comment (gsmet) [inline docs/src/main/asciidoc/extension-faq.adoc:35]:
```suggestion
Have a look at the xref:extension-maturity-matrix.adoc[extension maturity matrix].
```

File: docs/src/main/asciidoc/extension-faq.adoc [hunk 3]
```diff
@@ -19,8 +19,8 @@ The document header ends at the first blank line. Do not remove the blank line b
 ### Why would I want to write an extension?
 
 See the xref:writing-extensions#extension-philosophy[extension philosophy].
```
Comment (gsmet) [inline docs/src/main/asciidoc/extension-faq.adoc:21]:
```suggestion
See the xref:writing-extensions.adoc#extension-philosophy[extension philosophy].
```

File: docs/src/main/asciidoc/extension-faq.adoc [hunk 4]
```diff
@@ -19,8 +19,8 @@ The document header ends at the first blank line. Do not remove the blank line b
 ### Why would I want to write an extension?
```
Comment (gsmet) [inline docs/src/main/asciidoc/extension-faq.adoc:19]: Asciidoc should use `=` for titles not `#`.

**Review by gsmet — CHANGES_REQUESTED** [review]
What I pointed out might help getting the build green.

**Comment by holly-cummins** [comment]
> What I pointed out might help getting the build green.

Thanks! I've squashed, and then changed the styles of the headings in a second commit.

**Review by gsmet — APPROVED** [review]
Awesome, thanks!

**Pull request merged**